lib/internal/modules/esm/load.js

```diff
@@ -29,37 +29,42 @@ const {
 
 const DATA_URL_PATTERN = /^[^/]+\/[^,;]+(?:[^,]*?)(;base64)?,([\s\S]*)$/;
 
+/**
+ * @param {URL} url URL to the module
+ * @param {ESModuleContext} context used to decorate error messages
+ * @returns {{ responseURL: string, source: string | BufferView }}
+ */
 async function getSource(url, context) {
-  const parsed = new URL(url);
-  let responseURL = url;
+  const { protocol, href } = url;
+  let responseURL = href;
   let source;
-  if (parsed.protocol === 'file:') {
+  if (protocol === 'file:') {
     const { readFile: readFileAsync } = require('internal/fs/promises').exports;
-    source = await readFileAsync(parsed);
-  } else if (parsed.protocol === 'data:') {
-    const match = RegExpPrototypeExec(DATA_URL_PATTERN, parsed.pathname);
+    source = await readFileAsync(url);
```

Comment:
I assume it’s faster not to cast to a string if we don’t need to?

```suggestion
    source = await readFileAsync(href);
```

Reply:
It's actually the reverse--readFile accepts string or URL (and some other things...) and if it's a string it will convert it to a file URL object.